contributor/02-becoming-a-contributor.md

```diff
@@ -0,0 +1,45 @@
+
+In contrast to traditional project InnerSource projects come with the role of a Contributor: Yes, there will be times, when you wish that the shared solution has a new feature. As a Contributor you have the freedom to checkout the source code of the component, make modifications to it and use the resulting improved version.
+
+Yes, there will be times, when you urgently need a bug fix on your timeline that doesn't have the same priority in the host team. Becoming a Contributor enables you to become active yourself and support the host team with fixing that but.
+
+This way of working requires a change in mindset for many: Instead of waiting for features to be implemented or bugs fixed, instead of working around issues there's now a third solution. Spend your time and energy to check back with the InnerSource project on what your needs are - and then make the change directly in the shared project. So in addition to your coding skills, what you need to be successful in an InnerSource project are communication skills to clearly articulate your needs and come to a solution to address those that works for both, your team and the host team.
+
+"But I could simply go and fork the project, make my changes there and save myself from all this coordination overhead!" Sure - forking the project is a way to get your job done. Except in the long run this means it's on you to maintain that forked version for your team - and carry your changes forward for any new release the host team makes. Contributing your changes to the host team also means you get to benefit from their deeper knowledge of the component. They may spot issues in your patch, that otherwise would only have become obvious in production.
+
+A good Contributor can comfortably make a call for when it makes both, technical and business sense to introduce a dependency and re-use a component instead of duplicating work. They can talk to management to explain the benefits of InnerSource contributions.
+
+#### Scope of InnerSource contributions
+
+So is Inner*Source* only about *Source*Code? Of course not. If your team's business depends on an outside component you want to make sure it's well maintained and well run. As an InnerSource Contributor, you can help the host team with that: Reporting issues you see when using the component is a valuable contribution. Creating or fixing test cases that show that the code isn't working as expected is valuable. So is improving documentation so others spend less time using it and contributing to it. Supporting other users, helping with bug triage can be valuable contributions. Improving builds is another example of a valuable contribution. 
+
+To summarize no contribution is too small to contribute. Here is one that I made to tensorflow/models. A simple label change in a graph.
```

Comment:
If we mention the tensorflow/models fix we should actually link to it.

Reply:
https://github.com/tensorflow/models/pull/4784

Reply:
Not sure this part actually needs to be in there, though.

Reply:
I've found that examples from open source real world experiences helps make the story for inner source easier to understand. So I'm in favour of adding the link.

Reply:
Works for me.